src/ViewLayer/Faults/BatteryFaults/BatteryFaultList.cpp

```diff
@@ -32,13 +32,20 @@ FaultDisplayData BatteryFaultList::nextActiveFault()
     return FaultDisplayData();
 }
 
+QVector<QString>& BatteryFaultList::activeFaultLabels()
+{
+    numberOfActiveFaults();
+    return activeLabels_;
+}
+
 QMap<BatteryFaults, FaultDisplayData>& BatteryFaultList::faults()
 {
     return faultList_;
 }
 
-int BatteryFaultList::numberOfActiveFaults() const
+int BatteryFaultList::numberOfActiveFaults()
```

Comment:
The new changes in this function turns this function from just having one purpose (which is to just get the number of active faults), into having two purposes (get the number of active faults, and reassigning activeLabels_).

Could we have a another function that takes care of reassigning activeLabels_?